sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,489 @@ def reduce_inequalities(inequalities, symbols=[]):
+    standard_constraints = []
+    eqns = []
+
+    for rel in constraints:
+        if not isinstance(rel, (Relational, AppliedBinaryRelation)):
+            raise TypeError(f"{rel} is not relational.")
+        if type(rel) in [Lt, Gt] or isinstance(rel, AppliedBinaryRelation) and rel.function in [Q.lt, Q.gt]:
+            raise TypeError("Strict inequalities are not allowed in linear programming.")
+        if type(rel) == Ne or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.ne:
+            raise TypeError("'not equal to' is not allowed in linear programming.")
+
+        variables = variables | rel.free_symbols
+
+        if type(rel) == Le or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.le:
+            eqns.append(rel.lhs - rel.rhs)
+            standard_constraints.append(rel.lhs <= rel.rhs)
+        elif type(rel) == Ge or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.ge:
+            eqns.append(rel.rhs - rel.lhs)
+            standard_constraints.append(-rel.lhs <= -rel.rhs)
+        elif type(rel) == Eq or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.eq:
+            # x = 3 can be represented as x <= 3 and x >= 3
+            eqns.append(rel.lhs - rel.rhs)
+            standard_constraints.append(rel.lhs <= rel.rhs)
+            eqns.append(rel.rhs - rel.lhs)
+            standard_constraints.append(-rel.lhs <= -rel.rhs)
+        else:
+            raise TypeError(f"Unrecognized relation: {rel}")
+
+    variables = sorted(variables, key=lambda v: str(v)) # order symbols lexicographically
+
+    A, B = linear_eq_to_matrix(eqns, *variables)
+    C = linear_eq_to_matrix(objective, *variables)[0] # constant terms can be safely ignored here
+
+    return A, B, C, variables, standard_constraints
+
+def linprog_maximize(constraints, objective):
+    """
+    A function to maximize a linear objective function subject to linear
+    constraints with the simplex method.
+
+    Parameters
+    ==========
+
+    constraints : list of linear inequalities
+        Stict inequalities (>, <) and not equals are not allowed.
+
+    objective : a linear function to maximize
+        If you want to minimize a function, simply make the objective function
+        negative.
+
+    Returns
+    =======
+
+    Returns a tuple of solutions optimum, argmax:
+
+    optimum : Rational or Float
+
+    argmax : dictionary of variables to Rationals or Floats
+
+    argmax_dual : dictionary of constraints to Rationals or Floats
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.solvers.inequalities import linprog_maximize
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> optimum, argmax, argmax_dual  = linprog_maximize([r1,r2,r3], x+y+5*z)
+    >>> optimum
+    11/3
+    >>> argmax
+    {x: 0, y: 1/3, z: 2/3}
+    >>> argmax_dual
+    {y + 2*z <= 3: 0, -x - 3*z <= -2: 2/3, 2*x + y + 7*z <= 5: 1}
```

Comment:
The doctest is failing here because the key/values are in a different order. Not sure how to fix this. Maybe I should sort the inequalities before I convert them to a dictionary?

I'm also not sure if it's useful to even make this a dictionary. It might be better to just return the values or even not return argmax_dual in this function.

Reply:
I thought you have removed the usage of random. If you remove all source of nondeterminism (like random, set) then it will be fine.

The worst thing to do is to use sort inside the library code for not obvious reasons. If you can’t avoid that, it is better to sort before printing in the doctest.

Reply:
Random is indeed removed. 

Actually I think I can fix this by just changing the order. For some reason PyCharm was okay with this order but `bin/doctest --force-colors` doesn't like it.